confidence/models.py

```diff
@@ -108,9 +110,14 @@ def _resolve(self, value):
 
             return value
         except NotConfiguredError as e:
-            raise ConfiguredReferenceError(f'unable to resolve referenced key {match.group("path")}', key=e.key) from e
-
-    def get(self, path, default=_NoDefault, as_type=None, resolve_references=True):
+            missing_key = match.group('path')  # type: ignore
+            raise ConfiguredReferenceError(f'unable to resolve referenced key {missing_key}', key=e.key) from e
+
+    def get(self,
+            path: str,
+            default: typing.Any = NoDefault,
+            as_type: typing.Optional[typing.Callable] = None,
+            resolve_references: bool = True) -> typing.Any:
```

Comment:
Note for the future (particularly when we come around to creating proper docs): attribute access defaulting to 'safe fake values' and `get()` defaulting to raising errors is a bit reversed from the default dict-like operation of things in Python :thinking: